limiter/lifo_blocking.go

```diff
@@ -119,26 +119,29 @@ func (l *LifoBlockingListener) unblock() {
 	l.limiter.mu.Lock()
 	defer l.limiter.mu.Unlock()
 
-	if l.limiter.backlog.len() > 0 {
```

Comment:
Removing this length check means we dont have to check if the last item in the queue was evicted between `.len()` and `.peek()`. We can just rely on peek returning nil on empty queue